### PR TITLE
fix: add telemetry headers for appbase users only

### PR DESCRIPTION
**Pull request by mohdashraf010897**
**PR Type** `Bug`

Description: 
   - telemetry headers would throw CORS errors for non-appbase users.
   - https://github.com/appbaseio/appbase-js/pull/76#issuecomment-942436419

Test Cases:
image

### CODE REVIEW

File: src/core/api/getSuggestionsv3Api.js [hunk 1]
```diff
@@ -23,11 +23,22 @@ function getSuggestionsv3Api(query, settings) {
     settings: parsedSettings,
     query,
   };
+  const headers = {};
+
+  Object.assign(headers, {
```
Comment (bietkul) [inline src/core/api/getSuggestionsv3Api.js]: @mohdashraf010897 Use a utility method to get telemetry headers.

File: src/core/request/fetch.js [hunk 2]
```diff
@@ -9,20 +9,16 @@ import { btoa, removeUndefined } from '../../utils/index';
  * @param {String} args.path
```
Comment (bietkul) [inline src/core/request/fetch.js:9]: Why do we have changes in this file?

**Review by bietkul — CHANGES_REQUESTED** [review]
Share a demo, how did you test the telemetry header?

File: src/core/request/fetch.js (hunk 2 above)
Comment (mohdashraf010897) [reply src/core/request/fetch.js:9]: I needed to pass headers(`x-search-client`) that were specific to certain methods calling the fetch function, so added a key called `headers`

File: src/core/api/getSuggestionsv3Api.js (hunk 1 above)
Comment (mohdashraf010897) [reply src/core/api/getSuggestionsv3Api.js]: okay!

**Review by bietkul — APPROVED** [review]

**Pull request merged**